src/main/java/org/openstreetmap/atlas/geography/atlas/items/complex/aoi/ComplexAOI.java

```diff
@@ -0,0 +1,185 @@
+     *
+     * @param source
+     *            The {@link AtlasEntity} for which the ComplexEntity is created
+     * @param aoiFilter
+     *            The {@link TaggableFilter} of AOI tags against which the relation is checked for
+     *            AOI tags
+     * @return {@link ComplexAOI} if created, else return empty.
+     */
+    public static Optional<ComplexAOI> getComplexAOI(final AtlasEntity source,
+            final TaggableFilter aoiFilter)
+    {
+        try
+        {
+            if (defaultTaggableFilter == null)
+            {
+                computeDefaultFilter();
+            }
+            return (source instanceof Relation || source instanceof Area)
+                    && (hasAOITag(source) || aoiFilter.test(source))
+                            ? Optional.of(new ComplexAOI(source)) : Optional.empty();
+        }
+        catch (final Exception exception)
+        {
+            logger.warn("Unable to create complex AOI relations from {}", source, exception);
+            return Optional.empty();
+        }
+    }
+
+    private static void computeDefaultFilter()
+    {
+        try (InputStreamReader reader = new InputStreamReader(
+                ComplexAOI.class.getResourceAsStream(AOI_RESOURCE)))
+        {
+            final JsonElement element = new JsonParser().parse(reader);
+            final JsonArray filters = element.getAsJsonObject().get("filters").getAsJsonArray();
+            defaultTaggableFilter = StreamSupport.stream(filters.spliterator(), false)
+                    .map(jsonElement -> TaggableFilter.forDefinition(jsonElement.getAsString()))
+                    .collect(Collectors.toList());
+        }
+        catch (final Exception exception)
+        {
+            throw new CoreException(
+                    "There was a problem parsing aoi-tag-filter.json. Check if the JSON file has valid structure.",
+                    exception);
+        }
+    }
+
+    /**
+     * Checks for AOI tags in the object
+     *
+     * @param source
+     *            {@link AtlasEntity} that needs to be checked for AOI tags
+     * @return {@code true} if the source has AOI tags
+     */
+    private static boolean hasAOITag(final AtlasEntity source)
+    {
+        return defaultTaggableFilter.stream()
+                .anyMatch(taggableFilter -> taggableFilter.test(source));
+    }
+
+    /**
+     * Construct a {@link ComplexAOI}
+     *
+     * @param source
+     *            the {@link AtlasEntity} to construct the ComplexAoiRelation
+     */
+    private ComplexAOI(final AtlasEntity source)
+    {
+        super(source);
+        try
+        {
+            this.multiPolygon = RELATION_OR_AREA_TO_MULTI_POLYGON_CONVERTER.convert(source);
+        }
+        catch (final Exception exception)
+        {
+            setInvalidReason("Unable to convert Relation to MultiPolygon", exception);
+            throw new CoreException("Unable to convert Relation to MultiPolygon", exception);
```

Comment:
Would it help having the relation identifier here for debugging?

Reply:
This Exception is caught in the method [getComplexAOI](https://github.com/osmlab/atlas/blob/5d67cea598acb2f6dc6fd0ed8237e321ac46f204/src/main/java/org/openstreetmap/atlas/geography/atlas/items/complex/aoi/ComplexAOI.java#L63) and the Relation is logged there. Would you prefer to have Relation identifier here as well?

Reply:
Good point. As long as it's logged somewhere in the stack, that works for me :)